modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -0,0 +1,391 @@
+ * More details are in the <a href="https://issues.apache.org/jira/browse/IGNITE-21140">epic</a>.
+ */
+public class DisasterRecoveryManager implements IgniteComponent {
+    /** Logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(DisasterRecoveryManager.class);
+
+    /** Single key for writing disaster recovery requests into meta-storage. */
+    static final ByteArray RECOVERY_TRIGGER_KEY = new ByteArray("disaster.recovery.trigger");
+
+    private static final TableMessagesFactory MSG_FACTORY = new TableMessagesFactory();
+
+    private static final int TIMEOUT = 30;
+
+    private static final int CATCH_UP_THRESHOLD = 10;
+
+    /** Thread pool executor for async parts. */
+    private final ExecutorService threadPool;
+
+    /** Messaging service. */
+    private final MessagingService messagingService;
+
+    /** Meta-storage manager. */
+    final MetaStorageManager metaStorageManager;
+
+    /** Catalog manager. */
+    final CatalogManager catalogManager;
+
+    /** Distribution zone manager. */
+    final DistributionZoneManager dzManager;
+
+    /** Raft manager. */
+    private final Loza raftManager;
+
+    /** Watch listener for {@link #RECOVERY_TRIGGER_KEY}. */
+    private final WatchListener watchListener;
+
+    /**
+     * Map of operations, triggered by local node, that have not yet been processed by {@link #watchListener}. Values in the map are the
+     * futures, returned from the {@link #processNewRequest(ManualGroupUpdateRequest)}, they are completed by
+     * {@link #handleTriggerKeyUpdate(WatchEvent)} when node receives corresponding events from the metastorage (or if it doesn't receive
+     * this event within a 30 seconds window).
+     */
+    private final Map<UUID, CompletableFuture<Void>> ongoingOperationsById = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor.
+     */
+    public DisasterRecoveryManager(
+            ExecutorService threadPool,
+            MessagingService messagingService,
+            MetaStorageManager metaStorageManager,
+            CatalogManager catalogManager,
+            DistributionZoneManager dzManager,
+            Loza raftManager
+    ) {
+        this.threadPool = threadPool;
+        this.messagingService = messagingService;
+        this.metaStorageManager = metaStorageManager;
+        this.catalogManager = catalogManager;
+        this.dzManager = dzManager;
+        this.raftManager = raftManager;
+
+        watchListener = new WatchListener() {
+            @Override
+            public CompletableFuture<Void> onUpdate(WatchEvent event) {
+                return handleTriggerKeyUpdate(event);
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                // No-op.
+            }
+        };
+
+        messagingService.addMessageHandler(TableMessageGroup.class, this::handleMessage);
```

Comment:
Shouldn't it be in `#start`?

Reply:
I saw both patterns. Let's move it to start